src/background/dnr.js

```diff
@@ -9,41 +9,158 @@
+
+          try {
+            console.info('[dnr] Updating dynamic fixes rules...');
+
+            const list = await fetch(
+              `${ENGINE_CONFIGS_ROOT_URL}/dnr-fixes-v2/allowed-lists.json`,
+            ).then((res) =>
+              res.ok
+                ? res.json()
+                : Promise.reject(
+                    new Error(
+                      `Failed to fetch allowed lists: ${res.statusText}`,
+                    ),
+                  ),
+            );
+
+            if (list.dnr.checksum !== resources.checksums['dnr-fixes']) {
+              let addRules = await fetch(list.dnr.url).then((res) =>
+                res.ok
+                  ? res.json()
+                  : Promise.reject(
+                      new Error(`Failed to fetch DNR rules: ${res.statusText}`),
+                    ),
+              );
+
+              if (__PLATFORM__ === 'safari') {
+                addRules = addRules.reduce((acc, rule) => {
+                  try {
+                    acc.push(convertToSafariFormat(rule));
+                  } catch {
+                    // Ignore rules that cannot be converted
+                  }
+                  return acc;
+                }, []);
+              }
+
+              await chrome.declarativeNetRequest.updateDynamicRules({
+                removeRuleIds: await getDynamicRulesIds(FIXES_ID_RANGE),
+                addRules: addRules.map((rule, index) => ({
+                  ...rule,
+                  id: FIXES_ID_RANGE.start + index,
+                })),
+              });
+
+              console.info(
+                '[dnr] Updated dynamic fixes rules:',
+                list.dnr.checksum,
+              );
+              await store.set(resources, {
+                checksums: { [DNR_FIXES_KEY]: list.dnr.checksum },
+              });
+            }
+          } catch (e) {
+            console.error('[dnr] Error while updating dynamic fixes rules:', e);
+
+            // If no dynamic rules are applied, it means that initial fetch failed.
+            // As a fallback we need to add static fixes rules.
+            if (!removeRuleIds.length) {
+              console.warn('[dnr] Falling back to static fixes rules');
+              ids.push('fixes');
+
+              await store.set(resources, {
+                checksums: { [DNR_FIXES_KEY]: 'filesystem' },
+              });
+            }
+          }
+        }
+      } else if (resources.checksums[DNR_FIXES_KEY]) {
+        // Remove all dynamic fixes rules
+        const removeRuleIds = await getDynamicRulesIds(FIXES_ID_RANGE);
+        if (removeRuleIds.length) {
+          await chrome.declarativeNetRequest.updateDynamicRules({
+            removeRuleIds,
+          });
+          await store.set(resources, {
+            checksums: { [DNR_FIXES_KEY]: null },
+          });
+
+          console.info('[dnr] Removed dynamic fixes rules');
+        }
+      }
+    } else if (ids.length) {
+      ids.push('fixes');
```

Comment:
having a note that this is a default behavior for the first run may help in the future

Reply:
It won't be if we set a flag to 100%.